From: Anthony Sottile <asottile@umich.edu>
Date: Thu, 4 Jul 2019 18:45:29 -0700
Subject: fix docs build

---
 Doc/tools/extensions/pyspecific.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Doc/tools/extensions/pyspecific.py b/Doc/tools/extensions/pyspecific.py
index 91f932d..0f3536f 100644
--- a/Doc/tools/extensions/pyspecific.py
+++ b/Doc/tools/extensions/pyspecific.py
@@ -536,6 +536,7 @@ def process_audit_events(app, doctree, fromdocname):
         nodes.colspec(colwidth=30),
         nodes.colspec(colwidth=55),
         nodes.colspec(colwidth=15),
+        cols=3,
     )
     head = nodes.thead()
     body = nodes.tbody()
