From: Anthony Sottile <asottile@umich.edu>
Date: Mon, 2 Dec 2019 10:19:37 -0800
Subject: allow older sphinx

---
 Doc/conf.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Doc/conf.py b/Doc/conf.py
index a03324a..e8f656a 100644
--- a/Doc/conf.py
+++ b/Doc/conf.py
@@ -45,7 +45,7 @@ today_fmt = '%B %d, %Y'
 highlight_language = 'python3'
 
 # Minimum version of sphinx required
-needs_sphinx = '1.3'
+needs_sphinx = '1.2'
 
 # Ignore any .rst files in the venv/ directory.
 exclude_patterns = ['venv/*', 'README.rst']
